cmd/protoc-gen-connect-go/main.go

```diff
@@ -144,12 +168,37 @@ func generatePreamble(g *protogen.GeneratedFile, file *protogen.File) {
 }
 
 func generateServiceNameConstants(g *protogen.GeneratedFile, services []*protogen.Service) {
+	var numMethods int
 	g.P("const (")
 	for _, service := range services {
 		constName := fmt.Sprintf("%sName", service.Desc.Name())
 		wrapComments(g, constName, " is the fully-qualified name of the ",
 			service.Desc.Name(), " service.")
 		g.P(constName, ` = "`, service.Desc.FullName(), `"`)
+		numMethods += len(service.Methods)
+	}
+	g.P(")")
+	g.P()
+
+	if numMethods == 0 {
+		return
+	}
+	wrapComments(g, "These constants are the fully-qualified names of the RPCs defined in this package. ",
+		"They're exposed at runtime as Spec.Procedure and as the final two segments of the HTTP route.")
+	g.P("//")
+	wrapComments(g, "Note that these are different from the fully-qualified method names used by ",
```

Comment:
I don't love this either, but I don't know what to do

Reply:
Complain to the gRPC people? We're effectively stuck with this for compatibility. We can skip documenting the inconvenient fact that it's not the same as protobuf fully-qualified method names for some inscrutable reason, but that seems like a pretty significant footgun.